tools/optrack/optrack_to_t2.py

```diff
@@ -0,0 +1,513 @@
+            logfile.write("Could not find the matching operation begin record" +
+                          " for the following operation end record: \n");
+            logfile.write(str(intervalEnd) + "\n");
+            return None;
+        if (intervalBegin[2] != intervalEnd[2]):
+            logfile.write("Operation end record does not match the available " +
+                          "operation begin record. " +
+                          "Your log file may be incomplete.\n" +
+                          "Skipping the begin record.\n");
+            logfile.write("Begin: " + str(intervalBegin) + "\n");
+            logfile.write("End: " + str(intervalEnd) + "\n");
+            errorOccurred = True;
+        else:
+            matchFound = True;
+
+    return intervalBegin[0], intervalEnd[0], intervalEnd[2], errorOccurred;
+
+def createCallstackSeries(data, logfilename):
+
+    beginIntervals = [];
+    dataFrame = None;
+    endIntervals = [];
+    errorReported = False;
+    functionNames = [];
+    intervalBeginningsStack = [];
+    largestStackDepth = 0;
+    logfile = None;
+    thisIsFirstRow = True;
+
+    # Let's open the log file.
+    try:
+        logfile = open(logfilename, "w");
+    except:
+        logfile = sys.stdout;
+
+    for row in data.itertuples():
+        # row[0] is the timestamp, row[1] is the event type,
+        # row[2] is the function name.
+        #
+        if (row[1] == 0):
+            intervalBeginningsStack.append(row);
+        elif (row[1] == 1):
+            try:
+                intervalBegin, intervalEnd, function, error\
+                    = getIntervalData(intervalBeginningsStack, row, logfile);
+                if (error and (not errorReported)):
+                    errorReported = reportDataError(logfile, logfilename);
+            except:
+                if (not errorReported):
+                    errorReported = reportDataError(logfile, logfilename);
+                continue;
+
+            beginIntervals.append(intervalBegin);
+            endIntervals.append(intervalEnd);
+            functionNames.append(function);
+
+        else:
+            print("Invalid event in this line:");
+            print(str(row[0]) + " " + str(row[1]) + " " + str(row[2]));
+            continue;
+
+    if (len(intervalBeginningsStack) > 0):
+        logfile.write(str(len(intervalBeginningsStack)) + " operations had a " +
+                      "begin record, but no matching end records. " +
+                      "Please check that your operation tracking macros " +
+                      "are properly inserted.\n");
+        if (not errorReported):
+            errorReported = reportDataError(logfile, logfilename);
+        intervalBeginningsStack = [];
+
+    dict = {};
+    dict['start'] = beginIntervals;
+    dict['end'] = endIntervals;
+    dict['function'] = functionNames;
+    dict['stackdepth'] = [0] * len(beginIntervals);
+
+    dataframe = pd.DataFrame(data=dict);
```

Comment:
In my opinion, it would be better not to use `dict` as an identifier since it's the same name as Python's dictionary constructor.

For example, this would make a new dictionary.
```
dataframe = pd.DataFrame(data=dict())
```